lighthouse-cli/test/smokehouse/frontends/smokehouse-bin.js

```diff
@@ -65,6 +65,44 @@ function getDefinitionsToRun(allTestDefns, requestedIds, {invertMatch}) {
   return smokes;
 }
 
+/**
+ * Prune the `networkRequests` from the test expectations when `takeNetworkRequestUrls`
+ * is not defined. Custom servers may not have this method available in-process.
+ * Also asserts that any expectation with `networkRequests` is run serially. For core
+ * tests, we don't currently have a good way to map requests to test definitions if
+ * the tests are run in parallel.
+ * @param {Array<Smokehouse.TestDfn>} testDefns
+ * @param {Function|undefined} takeNetworkRequestUrls
+ * @return {Array<Smokehouse.TestDfn>}
+ */
+function pruneExpectedNetworkRequests(testDefns, takeNetworkRequestUrls) {
+  const pruneNetworkRequests = !takeNetworkRequestUrls;
+
+  const clonedDefns = cloneDeep(testDefns);
+  for (const {id, expectations, runSerially} of clonedDefns) {
+    for (const expectation of expectations) {
+      if (!runSerially && expectation.networkRequests) {
```

Comment:
In the current version, if a test is `runSerially: false` but has a `networkRequests` expectation, only a warning is printed when run locally, but it's a fatal error in CI. I believe this is due to my suggestions, since originally I was thinking we [always run smoke serially in CI with `-j=1`](https://github.com/GoogleChrome/lighthouse/blob/8b41c549f00488e6856a9c3992fe7ee92eb3e5d8/.github/workflows/smoke.yml#L53), so we'll always get the test one way or another.

I eventually came around to @adamraine's thinking that we should really be looking at only the individual tests's `runSerially` value so that there's no surprises depending on if running locally or in CI (and if running with `-j=1` or not), so this is just making that concrete. No one should add `networkRequests` expectations unless they're assertable in all environments :)